docusaurus/docs/cms/admin-panel-customization/homepage.md

```diff
@@ -0,0 +1,761 @@
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```tsx title="src/plugins/my-plugin/admin/src/components/MyWidget/index.tsx"
+import React, { useState, useEffect } from 'react';
+import { Widget } from '@strapi/admin/strapi-admin';
+
+interface DataItem {
+  id: number;
+  name: string;
+}
+
+const MyWidget: React.FC = () => {
+  const [loading, setLoading] = useState<boolean>(true);
+  const [data, setData] = useState<DataItem[] | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    // Fetch your data here
+    const fetchData = async () => {
+      try {
+        // Replace with your actual API call
+        const response = await fetch('/my-plugin/data');
+        const result = await response.json();
+        
+        setData(result);
+        setLoading(false);
+      } catch (err) {
+        setError(err instanceof Error ? err : new Error(String(err)));
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  if (loading) {
+    return <Widget.Loading />;
+  }
+
+  if (error) {
+    return <Widget.Error />;
+  }
+
+  if (!data || data.length === 0) {
+    return <Widget.NoData>No data available</Widget.NoData>;
+  }
+
+  return (
+    <div>
+      {/* Your widget content here */}
+      <ul>
+        {data.map((item) => (
+          <li key={item.id}>{item.name}</li>
+        ))}
+      </ul>
+    </div>
+  );
+};
+
+export default MyWidget;
+```
+
+</TabItem>
+</Tabs>
+
+**Data management**:
+
+![Rendering and Data management](/img/assets/homepage-customization/rendering-data-management.png)
+
+The green box above represents the area where the user’s React component (from `widget.component` in the [API](#widget-api-reference)) is rendered. You can render whatever you like inside of this box. Everything outside that box is, however, rendered by Strapi. This ensures overall design consistency within the admin panel. The `icon`, `title`, and `link` (optional) properties provided in the API are used to display the widget.
+
+#### Widget helper components reference
+
+Strapi provides several helper components to maintain a consistent user experience across widgets:
+
+| Component        | Description                                         | Usage                                |
+|------------------|-----------------------------------------------------|--------------------------------------|
+| `Widget.Loading` | Displays a loading spinner and message              | When data is being fetched           |
+| `Widget.Error`   | Displays an error state                             | When an error occurs                 |
+| `Widget.NoData`  | Displays when no data is available                  | When the widget has no data to show  |
+| `Widget.NoPermissions` | Displays when user lacks required permissions | When the user cannot access the widget |
+
+These components help maintain a consistent look and feel across different widgets.
```

Comment:
A common thing for all these components:
- you could render them without children and you'll get the default wording: `<Widget.Error />`
- or you could pass children to override our copy and specify your own wording: `<Widget.Error>Oopsie!</Widget.Error>`

Reply:
Thanks, I'll add this to docs.